src/main/java/org/apache/cassandra/sidecar/tasks/PeriodicTask.java

```diff
@@ -60,6 +60,11 @@ default TimeUnit initialDelayUnit()
      * Defines the task body.
      * The method can be considered as executing in a single thread.
      *
+     * <br><b>NOTE:</b> It is very important that the promise is completed (as either succeeded or failed), when the
```

Comment:
we need to add this note, to make sure implementing classes are aware of this detail